wasm scope proposal

zuzunker opened the pull request.

zuzunker committed:
tiny dapp example fix

zuzunker committed:
prototyping

zuzunker committed:
prototyping

zuzunker committed:
Merge remote-tracking branch 'origin/develop' into ruslan/csl-mem

# Conflicts:
#	packages/yoroi-ergo-connector/example-cardano/index.js

zuzunker committed:
Moving the `WasmScope` into the rust-module

zuzunker committed:
Extracting proxy handler as static per scope

zuzunker committed:
more comments

zuzunker committed:
Extracted scope-creating function for better readability

zuzunker committed:
Tiny renaming

zuzunker committed:
more comments

zuzunker committed:
Flow fixes

zuzunker committed:
Fixed async handling

zuzunker committed:
Adding wasm-scope usage

zuzunker committed:
flow fixes

zuzunker committed:
adding proper error handling

zuzunker committed:
rolling back unnecessary changes in imports

zuzunker committed:
flow fixes

zuzunker committed:
comment fix

neuodev commented on packages/yoroi-extension/app/api/ada/lib/cardanoCrypto/rustLoader.js:
```diff
@@ -13,6 +17,113 @@ const MAX_TX_BYTES = 16384;
+}
+
+function isPromise(o: ?any): boolean {
+  return o != null && (typeof o.then === 'function') && (typeof o.catch === 'function');
```
```suggestion
  return o instanceof Promise;
```

zuzunker replied:
Makes sense!

zuzunker replied:
Removed the function completely now 👌

neuodev commented:
I was trying to write some unit testing 

```js
   // rustLoader.test.js
    it('Test...', () => {
        const addr = 'addr1qf2w3pe8jsyvr9kutpv6e0rzuaym7dvq0rrz9699a7y3wwvz6g3gz764vxla692p4ttcekdw0smpedxuqq7a8t8sv3usrcv98z';
        const fixedAddr = RustModule.WasmScope(Module => bytesToHex(
          Module.WalletV4.Address.from_bech32(addr).to_bytes()
        ));

        console.log(fixedAddr)
    })
```

But I got the `Maximum call stack size exceeded` from the recursive function!

zuzunker committed:
fixes promise check

zuzunker committed:
proxy check fixed

zuzunker committed:
unit test added

zuzunker commented:
> But I got the `Maximum call stack size exceeded` from the recursive function!

@AhmedIbrahim336 , weird, call stack should not be happening. I have added some fixes, and also added a unit test now, plz check again: https://github.com/Emurgo/yoroi-frontend/pull/3093/commits/6056e156a8c02d9c8e83b00caf5cd4922bf48a55

UPD: latest version of the test is here https://github.com/Emurgo/yoroi-frontend/blob/c9a6df5413a679b051fff42d8431e1d8de02c9bb/packages/yoroi-extension/app/api/ada/lib/cardanoCrypto/rustLoader.test.js

zuzunker committed:
unused imports removed

zuzunker committed:
fixed wasm check and added internal api for tests

neuodev commented:
> @AhmedIbrahim336 , weird, call stack should not be happening. I have added some fixes, and also added a unit test now, plz check again: [6056e15](https://github.com/Emurgo/yoroi-frontend/commit/6056e156a8c02d9c8e83b00caf5cd4922bf48a55)

Nice, I will check it!

zuzunker committed:
Merge branch 'develop' into ruslan/csl-mem

zuzunker committed:
Merge branch 'develop' into ruslan/csl-mem

zuzunker reviewed (commented):
/check

zuzunker committed:
Merge branch 'develop' into ruslan/csl-mem
